src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,95 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Comment } from '../types/Comment';
+import { addComment } from '../api/posts';
+
+type Props = {
+  activePostId: number,
+  postComments: Comment[],
+  setPostComments(comment: Comment[]): void,
+};
+
+interface NewComment {
+  name: string;
+  email: string;
+  body: string;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  activePostId,
+  postComments,
+  setPostComments,
+}) => {
+  const [newComment, setNewComment] = useState<NewComment>({
+    name: '',
+    email: '',
+    body: '',
+  });
+  const [isAdding, setIsAdding] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isTextError, setIsTextError] = useState(false);
+  const addPostComments = async (comment: Omit<Comment, 'id'>) => {
```

Comment:
```suggestion
  const [isTextError, setIsTextError] = useState(false);

  const addPostComments = async (comment: Omit<Comment, 'id'>) => {
```
to increase readability